src/java/htsjdk/samtools/CRAMFileReader.java

```diff
@@ -99,36 +92,43 @@ public CRAMFileReader(final File cramFile, final InputStream inputStream,
      *                        reference sequences
      */
     public CRAMFileReader(final File cramFile, final File indexFile,
-                          final ReferenceSource referenceSource) {
-        if (cramFile == null)
-            throw new IllegalArgumentException("File is required.");
-
-        this.cramFile = cramFile;
-        this.mIndexFile = indexFile;
-        this.referenceSource = referenceSource;
+                          final ReferenceSource referenceSource) throws FileNotFoundException,IOException {
+        this(new FileInputStream(cramFile),cramFile, new SeekableFileStream(indexFile), referenceSource, null);
+    }
 
-        getIterator();
+    public CRAMFileReader(final File cramFile, final ReferenceSource referenceSource) throws FileNotFoundException,IOException {
+        this(new FileInputStream(cramFile),cramFile, (SeekableStream)null, referenceSource, null);
     }
 
-    public CRAMFileReader(final File cramFile, final ReferenceSource referenceSource) {
-        if (cramFile == null && inputStream == null)
-            throw new IllegalArgumentException(
-                    "Either file or input stream is required.");
+    public CRAMFileReader(final InputStream inputStream, final SeekableStream indexInputStream,
+                          final ReferenceSource referenceSource, final ValidationStringency validationStringency) throws IOException {
+        this(inputStream,(File)null,indexInputStream,referenceSource,validationStringency);
+    }
 
+    CRAMFileReader(final InputStream inputStream,final File cramFile, final SeekableStream indexInputStream,
+                          final ReferenceSource referenceSource, final ValidationStringency validationStringency) throws IOException {
         this.cramFile = cramFile;
-        this.referenceSource = referenceSource;
 
-        getIterator();
-    }
+        if (inputStream != null) {
+        this.inputStream = inputStream;  
+        } else {
+            if (cramFile != null) {
+                this.inputStream = new FileInputStream(cramFile);            
+            } else {
+                throw new RuntimeException("inputStream or cramFile must not be null");
+            }
+        }
 
-    public CRAMFileReader(final InputStream inputStream, final SeekableStream indexInputStream,
-                          final ReferenceSource referenceSource, final ValidationStringency validationStringency) throws IOException {
-        this.inputStream = inputStream;
         this.referenceSource = referenceSource;
-        this.validationStringency = validationStringency;
 
-        iterator = new CRAMIterator(inputStream, referenceSource);
-        iterator.setValidationStringency(validationStringency);
+        iterator = new CRAMIterator(this.inputStream, referenceSource);
+
+        if (validationStringency != null) {
```

Comment:
what does this null check achieve? If validationStringency is null then this.validationStringency will also remain null so is this a no-op?